RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneFlatten.h

```diff
@@ -105,13 +201,62 @@ class HiEvtPlaneFlatten {
     return psi;
   }
   
+  double GetOffsetPsi(double s, double c, double w, uint m,  double vtx, int centbin)
+  {
+    int indx = GetOffsetIndx(centbin,vtx);
+    double snew = s-yoffDB[indx];
+    double cnew = c-xoffDB[indx];
+    double psi = atan2(snew,cnew)/vorder;
+    if((fabs(snew)<1e-4) && (fabs(cnew)<1e-4)) psi = 0.;
+    psi=bounds(psi);
+    psi=bounds2(psi);
+    soff_ = snew;
+    coff_ = cnew;
```

Comment:
this is the only place where soff_ and coff_ are initialized to any value;
if GetOffsetPsi is never called, their values are undefined.
Better add to the constructor.